kotlin/vci-client/src/main/java/io/mosip/vciclient/authorizationCodeFlow/interactiveAuthorization/presentationDuringIssuance/PresentationDuringIssuanceAuthorizationMethodService.kt

```
@@ -0,0 +1,278 @@
+import io.mosip.openID4VP.authorizationResponse.unsignedVPToken.UnsignedVPToken
+import io.mosip.openID4VP.authorizationResponse.vpTokenSigningResult.VPTokenSigningResult
+import io.mosip.openID4VP.constants.FormatType
+import io.mosip.openID4VP.constants.ResponseMode
+import io.mosip.vciclient.authorizationCodeFlow.interactiveAuthorization.handler.AuthorizationMethodService
+import io.mosip.vciclient.authorizationCodeFlow.interactiveAuthorization.request.AuthorizationRequestData
+import io.mosip.vciclient.authorizationCodeFlow.interactiveAuthorization.response.AuthorizationResponse
+import io.mosip.vciclient.common.JsonUtils
+import io.mosip.vciclient.exception.InteractiveAuthorizationException
+import io.mosip.vciclient.networkManager.HttpMethod
+import io.mosip.vciclient.networkManager.NetworkManager
+import io.mosip.vercred.vcverifier.keyResolver.types.did.DidPublicKeyResolver
+import kotlinx.coroutines.withTimeout
+
+class PresentationDuringIssuanceAuthorizationMethodService(
+    private val handlePresentationRequest: suspend (ovpRequest: AuthorizationRequest) -> Map<String, Map<FormatType, List<Any>>>,
+    private val signVerifiablePresentation: suspend (
+        payload: Map<FormatType, UnsignedVPToken>,
+    ) -> Map<FormatType, VPTokenSigningResult>,
+    private val openId4vp: OpenID4VP = OpenID4VP(traceabilityId = "", walletMetadata = null),
+    private val didPublicKeyResolver: DidPublicKeyResolver = DidPublicKeyResolver(),
+    private val handlePresentationTimeoutMs: Long = 500 * 1000L,
+    private val signVPTokensTimeoutMs: Long = 5 * 1000L
+) : AuthorizationMethodService {
+
+    override fun type(): String = "openid4vp_presentation"
+
+    override suspend fun authorizeUser(requestData: AuthorizationRequestData): AuthorizationResponse {
+        if (requestData !is OpenId4VpPresentationAuthorizationRequestData) {
+            return errorResponse("invalid_request", "Expected PresentationAuthorizationRequestData")
```

Instead of returning error response can an error be thrown from here?

Considering that AuthorizationResponse is built from the response received from auth server